altlas/Assets/scripts/MapGenerator.cs

```diff
@@ -290,11 +291,65 @@ public void spawnStacksInRow(float horLength, float verLength, List<List<MapData
         }
     }
 
-    /**
-     * spawnStackInRow with fixed drawer length
-     * */
-    public void spawnRowInDrawer(MapData[][] stacks, string category) {
-        spawnStacksInRow(DRAWER_HOR_LEN, DRAWER_VER_LEN, stacks, category);
+    void setLabelOnDraw(KeyValuePair<string, Dictionary<string, List<MapData>>> category)
+    {     
+        GameObject go = new GameObject("myText");
+        go.AddComponent<TextMesh>();
+        TextMesh tm = go.GetComponent(typeof(TextMesh)) as TextMesh;
+        go.name = category.Key + "_label";
+        var drawer = getDrawerObjectFromCategory(category.Key);
+        go.transform.parent = drawer.transform;
+
+        tm.text = categoryText[category.Key];
+        tm.anchor = TextAnchor.UpperCenter;
+        tm.alignment = TextAlignment.Center;
+        tm.transform.localScale = drawer.transform.Find("knob").transform.localScale;
+        tm.transform.localScale *= tm.transform.localScale.x * 0.25f;
+        tm.transform.position = drawer.transform.Find("knob").transform.position + new Vector3(0f, 0.08f, 0);
+        tm.transform.Rotate(new Vector3(0, 1, 0), 180);
+        tm.font = Resources.GetBuiltinResource(typeof(Font), "Arial.ttf") as Font;
+        tm.font.material.shader = Shader.Find("Custom/Text Shader");
+    }
+
+    void setLabelOnStack()
```

Comment:
same change requests here.
- set the GameObject name using the constructor
- directly store the textmesh
- don't copy the knob's scale